raster/r.terraflow/types.h

```diff
@@ -26,8 +26,8 @@
 /* input parameters type */
 /* ------------------------------------------------------------ */
 
-typedef short dimension_type; /* represent dimension of the grid */
-static const dimension_type dimension_type_max=SHRT_MAX;
+typedef long dimension_type; /* represent dimension of the grid */
```

Comment:
using long is not a good idea: 1. the GRASS raster library does not support number of rows or cols to be long, they must be int, 2. long != long, it depends on the OS, e.g. on MS Windows 64 bit, long is 32 bit, while on Linux and UNIX 64 bit, long is 64 bit, and on Linux and UNIX 32 bit, long is 32 bit: use int.

Reply:
Thanks for the hint. Will try that.